### ISSUE TITLE
[BUG] The inferred type cannot be named without a reference to '../node_modules/inngest/components/InngestFunction' when using `composite` 

### ISSUE THREAD

**Issue by MonsterDeveloper** [inngest/inngest-js#384]
**Describe the bug**
When enabling the `composite` in a project's `tsconfig.json`, the compiler shows an error when using `inngest.createFunction`:

```
The inferred type of 'default' cannot be named without a reference to '../node_modules/inngest/components/InngestFunction'. This is likely not portable. A type annotation is necessary. typescript(2742)
```

**To Reproduce**

1. Create a function with `inngest.createFunction`
2. Enable `composite` in `tsconfig.json`

Link to reproduction: [CodeSandbox](https://codesandbox.io/p/sandbox/determined-sea-dd9c5w?layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522cloobmkiw00083b6e6ia1od1z%2522%252C%2522sizes%2522%253A%255B76.21592148309705%252C23.78407851690295%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522cloobmkiw00033b6ef8nhsajv%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522cloobmkiw00053b6egt2433r5%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522cloobmkiw00073b6e6oaomumz%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522cloobmkiw00033b6ef8nhsajv%2522%253A%257B%2522id%2522%253A%2522cloobmkiw00033b6ef8nhsajv%2522%252C%2522activeTabId%2522%253A%2522cloobojg600tu3b6eb50dtp7j%2522%252C%2522tabs%2522%253A%255B%257B%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fsrc%252Fclient.ts%2522%252C%2522id%2522%253A%2522cloobn6k400cp3b6eb6m3qfwq%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%252C%257B%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fsrc%252Ffunction.ts%2522%252C%2522id%2522%253A%2522cloobojg600tu3b6eb50dtp7j%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%255D%257D%252C%2522cloobmkiw00073b6e6oaomumz%2522%253A%257B%2522id%2522%253A%2522cloobmkiw00073b6e6oaomumz%2522%252C%2522tabs%2522%253A%255B%255D%257D%252C%2522cloobmkiw00053b6egt2433r5%2522%253A%257B%2522id%2522%253A%2522cloobmkiw00053b6egt2433r5%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522cloobmkiw00043b6e7nuv787v%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522dev%2522%257D%255D%252C%2522activeTabId%2522%253A%2522cloobmkiw00043b6e7nuv787v%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D)

**Expected behavior**
Inngest library should be working within composite projects without errors.

**Code snippets / Logs / Screenshots**
![CleanShot 2023-11-07 at 13 53 39@2x](https://github.com/inngest/inngest-js/assets/16190582/711f1544-6056-4707-804c-ab117f2260e2)

**System info (please complete the following information):**

- `"inngest": "3.4.2"`
- `"typescript": "^5.0.2"`

**Comment by jpwilliams** [comment]
Are yu using `pnpm`, @MonsterDeveloper? This might be related to microsoft/TypeScript#47663.

**Comment by MonsterDeveloper** [comment]
@jpwilliams after a bit of research, it seems to me that it is indeed an issue with the TypeScript compiler. I found a temporary fix — importing the types from the error directly:

```ts
import { InngestFunction } from "../node_modules/inngest/components/InngestFunction";
import type * as InngestTypes from "../node_modules/inngest/types";

// no errors now
export default inngest.createFunction(
  { id: "send-welcome-email" },
  { event: "user/new.signup" },
  async () => {
    return "some action";
  },
);
```
[CodeSandbox link](https://codesandbox.io/p/sandbox/pensive-varahamihira-k6p4zy?file=%2Fsrc%2Ffunction.ts%3A1%2C1&layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522cloobmkiw00083b6e6ia1od1z%2522%252C%2522sizes%2522%253A%255B76.21592148309705%252C23.78407851690295%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522cloobmkiw00033b6ef8nhsajv%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522cloobmkiw00053b6egt2433r5%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522cloobmkiw00073b6e6oaomumz%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522cloobmkiw00033b6ef8nhsajv%2522%253A%257B%2522id%2522%253A%2522cloobmkiw00033b6ef8nhsajv%2522%252C%2522activeTabId%2522%253A%2522cloobojg600tu3b6eb50dtp7j%2522%252C%2522tabs%2522%253A%255B%257B%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fsrc%252Fclient.ts%2522%252C%2522id%2522%253A%2522cloobn6k400cp3b6eb6m3qfwq%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%252C%257B%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fsrc%252Ffunction.ts%2522%252C%2522id%2522%253A%2522cloobojg600tu3b6eb50dtp7j%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%252C%257B%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Ftsconfig.json%2522%252C%2522id%2522%253A%2522clooqtx5q00823b6f64zgumcx%2522%252C%2522mode%2522%253A%2522temporary%2522%257D%255D%257D%252C%2522cloobmkiw00073b6e6oaomumz%2522%253A%257B%2522id%2522%253A%2522cloobmkiw00073b6e6oaomumz%2522%252C%2522tabs%2522%253A%255B%255D%257D%252C%2522cloobmkiw00053b6egt2433r5%2522%253A%257B%2522id%2522%253A%2522cloobmkiw00053b6egt2433r5%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522cloobmkiw00043b6e7nuv787v%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522dev%2522%257D%255D%252C%2522activeTabId%2522%253A%2522cloobmkiw00043b6e7nuv787v%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D)

But because `inngest` package doesn't export the types from `types` folder and `components/InngestFunction`, they have to be imported through a nasty `node_modules` direct import.

Since we can't fix `tsc` here, it'd be nice to include these exports so a better workaround can be used.

**Comment by jpwilliams** [comment]
Released in <kbd>[inngest@3.4.3](https://github.com/inngest/inngest-js/releases/tag/inngest%403.4.3)</kbd>.

Thanks for the thorough investigation and workaound, @MonsterDeveloper. 🙌
